tests/interpolation/test_interpolation.py

```diff
@@ -402,39 +369,53 @@ def test_interpolation_type_validated_error(
 
 
 @mark.parametrize(
-    ("cfg", "key"),
+    ("cfg", "key", "expected_value", "expected_node_type"),
     [
-        param({"dict": "${identity:{a: 0, b: 1}}"}, "dict.a", id="dict"),
         param(
-            {"dict": "${identity:{a: 0, b: {c: 1}}}"},
-            "dict.b.c",
-            id="dict_nested",
+            MissingList(list=SI("${identity:[0, 1, 2]}")),
+            "list",
+            [0, 1, 2],
+            ListConfig,
+            id="list_int_to_str",
+        ),
+        param(
+            MissingDict(dict=SI("${identity:{a: 0, b: 1}}")),
+            "dict",
+            {"a": 0, "b": 1},
+            DictConfig,
+            id="dict_int_to_str",
+        ),
+        param(
+            SubscriptedList(list=SI("${identity:[a, b]}")),
+            "list",
+            ["a", "b"],
+            ListConfig,
+            id="list_type_mismatch",
+        ),
+        param(
+            MissingDict(dict=SI("${identity:{0: b, 1: d}}")),
+            "dict",
+            {0: "b", 1: "d"},
+            DictConfig,
+            id="dict_key_type_mismatch",
```

Comment:
confused by those. why are they not expecting list and dict objects?

Reply:
I took a shortcut here as I was planning to keep the DictConfig / ListConfig but only after introducing `oc.create`, so in this commit the test is partially broken (I commented out the type check) and it's fixed in the next commit. Sorry it's a bit confusing.